ckanext/saml2/plugin.py

```diff
@@ -30,6 +30,13 @@ def user_create(context, data_dict):
 @logic.auth_sysadmins_check
 def user_update(context, data_dict):
     """Deny user changes."""
+    current_user = context['auth_user_obj']
+    if is_staff_user(current_user):
```

Comment:
I think for the general use case, the function should be `is_local_user()` or `not is_sso_user()`. I was thinking that with a User Principal Name as the SAML2 NameID CKAN could force 'local' users to have a username of 'username@localhost'  so the general test would be that a username with any domain besides @localhost must be an SSO user. It would prevent local users from using their email address as a username though, so perhaps not the best idea.  Maybe a configurable list of the domains that are SSO domains would be easier?